bug(lib): Random Provider doesn't build

<!--- Please keep this note for the community --->

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### cdktf & Language Versions

- 0.8 / 0.7 / potentially other versions
- Java

### Affected Resource(s)

Random provider

### Debug Output

<details>
🤖 build » compile | jsii --silence-warnings=reserved-word --no-fix-peer-dependencies
warning JSII6: A "peerDependency" on "cdktf" at "^0.8" means you should take a "devDependency" on "cdktf" at "0.8.0" (found "^0.8")"
warning JSII6: A "peerDependency" on "constructs" at "^10.0.0" means you should take a "devDependency" on "constructs" at "10.0.0" (found "^10.0.0")"
src/id.ts:105:14 - warning JSII5019: The property name "id" conflicts with the declaring class "Id". This will result in renaming the class to "_Id" in C#. Consider renaming "id".

105   public get id() {
                 ~~

  src/id.ts:33:14
    33 export class Id extends cdktf.TerraformResource {
                    ~~
    The declaring class is introduced here
🤖 build » compile » docgen | env: PATH=/Users/sebastian/.nv...
🤖 build » compile » docgen | jsii-docgen
Creating assembly in ts for @cdktf/provider-random from /var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/dGPtap (loose: true)
🤖 build » package | env: PATH=/Users/sebastian/.nv...
🤖 build » package | jsii-pacmak
[jsii-pacmak] [WARN] Exception occurred, not cleaning up /var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/npm-packgI52YU
[jsii-pacmak] [WARN] java failed
Error: All attempts failed. Last error: Command (mvn --quiet --batch-mode deploy -D=altDeploymentRepository=local::default::file:///var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/npm-packc85e58 --settings=user.xml) failed with status 1:
#STDERR>
#STDOUT> [ERROR] COMPILATION ERROR :
#STDOUT> [ERROR] /private/var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/npm-packgI52YU/_cdktf_provider-random/src/main/java/com/hashicorp/cdktf/providers/random_provider/$Module.java:[21,14] com.hashicorp.cdktf.providers.random_provider.$Module is not abstract and does not override abstract method resolveClass(java.lang.String) in software.amazon.jsii.JsiiModule
#STDOUT> [ERROR] /private/var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/npm-packgI52YU/_cdktf_provider-random/src/main/java/com/hashicorp/cdktf/providers/random_provider/$Module.java:[33,50] incompatible types: java.lang.String[] cannot be converted to com.hashicorp.cdktf.providers.random_provider.String[]
#STDOUT> [ERROR] /private/var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/npm-packgI52YU/_cdktf_provider-random/src/main/java/com/hashicorp/cdktf/providers/random_provider/$Module.java:[56,5] method does not override or implement a method from a supertype
#STDOUT> [ERROR] /private/var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/npm-packgI52YU/_cdktf_provider-random/src/main/java/com/hashicorp/cdktf/providers/random_provider/$Module.java:[70,34] incompatible types: com.hashicorp.cdktf.providers.random_provider.String cannot be converted to java.lang.String
#STDOUT> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project cdktf-provider-random: Compilation failure: Compilation failure:
#STDOUT> [ERROR] /private/var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/npm-packgI52YU/_cdktf_provider-random/src/main/java/com/hashicorp/cdktf/providers/random_provider/$Module.java:[21,14] com.hashicorp.cdktf.providers.random_provider.$Module is not abstract and does not override abstract method resolveClass(java.lang.String) in software.amazon.jsii.JsiiModule
#STDOUT> [ERROR] /private/var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/npm-packgI52YU/_cdktf_provider-random/src/main/java/com/hashicorp/cdktf/providers/random_provider/$Module.java:[33,50] incompatible types: java.lang.String[] cannot be converted to com.hashicorp.cdktf.providers.random_provider.String[]
#STDOUT> [ERROR] /private/var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/npm-packgI52YU/_cdktf_provider-random/src/main/java/com/hashicorp/cdktf/providers/random_provider/$Module.java:[56,5] method does not override or implement a method from a supertype
#STDOUT> [ERROR] /private/var/folders/zl/z8nc41qs7010qsfngfc_z1zc0000gn/T/npm-packgI52YU/_cdktf_provider-random/src/main/java/com/hashicorp/cdktf/providers/random_provider/$Module.java:[70,34] incompatible types: com.hashicorp.cdktf.providers.random_provider.String cannot be converted to java.lang.String
#STDOUT> [ERROR] -> [Help 1]
#STDOUT> [ERROR]
#STDOUT> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
#STDOUT> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
#STDOUT> [ERROR]
#STDOUT> [ERROR] For more information about the errors and possible solutions, please read the following articles:
#STDOUT> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
#STDOUT>
    at retry (/Users/sebastian/projects/cdk/provider/cdktf-provider-random/node_modules/jsii-pacmak/lib/util.js:70:27)
    at async Java.build (/Users/sebastian/projects/cdk/provider/cdktf-provider-random/node_modules/jsii-pacmak/lib/targets/java.js:303:9)
    at async /Users/sebastian/projects/cdk/provider/cdktf-provider-random/node_modules/jsii-pacmak/lib/targets/java.js:60:17
    at async Function.make (/Users/sebastian/projects/cdk/provider/cdktf-provider-random/node_modules/jsii-pacmak/lib/util.js:186:36)
    at async JavaBuilder.buildModules (/Users/sebastian/projects/cdk/provider/cdktf-provider-random/node_modules/jsii-pacmak/lib/targets/java.js:58:35)
    at async Promise.all (index 2)
    at async Object.pacmak (/Users/sebastian/projects/cdk/provider/cdktf-provider-random/node_modules/jsii-pacmak/lib/index.js:56:9)
🤖 Task "build » package" failed when executing "jsii-pacmak" (cwd: /Users/sebastian/projects/cdk/provider/cdktf-provider-random)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</details>

--- a/packages/@cdktf/provider-generator/lib/get/generator/resource-parser.ts
+++ b/packages/@cdktf/provider-generator/lib/get/generator/resource-parser.ts
@@ -24,6 +24,10 @@ const uniqueClassName = (className: string): string => {
   return className;
 };
 
+const isReservedClassName = (className: string): boolean => {
+  return ["string"].includes(className.toLowerCase());
+};
+
 class Parser {
   private structs = new Array<Struct>();
 
@@ -51,6 +55,10 @@ class Parser {
         optional: true,
         computed: false,
       };
+    }
+
+    if (isReservedClassName(baseName)) {
+      baseName = `${baseName}_resource`;
     }
 
     const className = uniqueClassName(toPascalCase(baseName));

--- a/test/java/synth-app/Main.java
+++ b/test/java/synth-app/Main.java
@@ -9,6 +9,9 @@ import com.hashicorp.cdktf.Testing;
 import imports.nullprovider.NullProvider;
 import imports.nullprovider.Resource;
 
+import imports.random.RandomProvider;
+import imports.random.StringResource;
+
 public class Main extends TerraformStack
 {
     public Main(final Construct scope, final String id) {
@@ -16,6 +19,9 @@ public class Main extends TerraformStack
 
         NullProvider.Builder.create(this, "Null").build();
         Resource resource = Resource.Builder.create(this, "NullResource").build();
+
+        RandomProvider.Builder.create(this, "Random").build();
+        StringResource stringResource = StringResource.Builder.create(this, "RandomString").length(42).build();
     }
 
     public static void main(String[] args) {

--- a/test/java/synth-app/cdktf.json
+++ b/test/java/synth-app/cdktf.json
@@ -1,7 +1,7 @@
 {
   "language": "java",
   "app": "mvn -e -q compile exec:java",
-  "terraformProviders": ["hashicorp/null@~> 3.1.0"],
+  "terraformProviders": ["hashicorp/null@~> 3.1.0", "hashicorp/random@~> 3.1.0"],
   "context": {
     "excludeStackIdFromLogicalIds": "true"
   }

